mplib/sapien_utils/conversion.py

```diff
@@ -196,10 +222,93 @@ def convert_sapien_col_shape(
         elif isinstance(shape, PhysxCollisionShapeSphere):
             collision_geom = Sphere(radius=shape.radius)
         elif isinstance(shape, PhysxCollisionShapeTriangleMesh):
-            # NOTE: see mplib.pymp.fcl.Triangle
-            raise NotImplementedError(
-                "Support for TriangleMesh collision is not implemented yet."
-            )
+            collision_geom = BVHModel()
+            collision_geom.begin_model()
+            collision_geom.add_sub_model(vertices=shape.vertices, faces=shape.triangles)
+            collision_geom.end_model()
         else:
             raise TypeError(f"Unknown shape type: {type(shape)}")
-        return [CollisionObject(collision_geom, pose.p, pose.q)]
+        return [CollisionObject(collision_geom, pose.p, pose.q)]  # type: ignore
+
+
+class SapienPlanner(Planner):
+    def __init__(
+        self,
+        sapien_planning_world: SapienPlanningWorld,
+        move_group: str,
+        *,
+        joint_vel_limits: Optional[Sequence[float] | np.ndarray] = None,
+        joint_acc_limits: Optional[Sequence[float] | np.ndarray] = None,
+    ):
+        """
+        Creates an mplib.planner.Planner from a SapienPlanningWorld.
+
+        :param sapien_planning_world: a SapienPlanningWorld created from sapien.Scene
+        :param move_group: name of the move group (end effector link)
+        :param joint_vel_limits: joint velocity limits for time parameterization
+        :param joint_acc_limits: joint acceleration limits for time parameterization
+        """
+        self.planning_world = sapien_planning_world
+        self.acm = self.planning_world.get_allowed_collision_matrix()
+
+        if len(planned_arts := self.planning_world.get_planned_articulations()) != 1:
+            raise NotImplementedError("Must have exactly one planned articulation")
+        self.robot = planned_arts[0]
+        self.pinocchio_model = self.robot.get_pinocchio_model()
+        self.user_link_names = self.pinocchio_model.get_link_names()
+        self.user_joint_names = self.pinocchio_model.get_joint_names()
+
+        self.joint_name_2_idx = {}
+        for i, joint in enumerate(self.user_joint_names):
+            self.joint_name_2_idx[joint] = i
+        self.link_name_2_idx = {}
+        for i, link in enumerate(self.user_link_names):
+            self.link_name_2_idx[link] = i
+
+        assert (
+            move_group in self.user_link_names
+        ), f"end-effector not found as one of the links in {self.user_link_names}"
+        self.move_group = move_group
+        self.robot.set_move_group(self.move_group)
+        self.move_group_joint_indices = self.robot.get_move_group_joint_indices()
+
+        self.joint_types = self.pinocchio_model.get_joint_types()
+        self.joint_limits = np.concatenate(self.pinocchio_model.get_joint_limits())
+        if joint_vel_limits is None:
```

Comment:
can we get this from sapien actually?

Reply:
Great question. Not completely.
If the urdf contains mimic joints, the pinocchio_model will not work properly (hence, SapienPlanner won't work).
This is because SAPIEN currently ignores the mimic joints and users need to implement special joints to handle them, making it impossible to know which joints are mimic joints during `export_kinematic_chain_urdf()`. 
I believe Fanbo will add a tag to make it possible to know the mimicked joints from a sapien.Scene.